src/main/scala/com/microsoft/hyperspace/index/IndexLogManager.scala

```diff
@@ -71,7 +71,7 @@ class IndexLogManagerImpl(indexPath: Path) extends IndexLogManager with Logging
       return None
     }
     val contents = FileUtils.readContents(fs, path)
-    Some(LogEntry.fromJson(contents))
+    Try(LogEntry.fromJson(contents)).toOption
```

Comment:
If `fromJson` throws a non-fatal exception, this will return `None` instead of throwing an exception, and thus cannot differentiate it with the `!fs.exists` case above. Can we revert this change for now? We can come back to this if required.

Reply:
So we want to catch only Non-fatal exceptions here ?